spec/controllers/secondary_content_links_controller_spec.rb

```diff
@@ -39,6 +39,22 @@
       expect(step_by_step_page.secondary_content_links.first.title).to eq("A Title")
     end
 
+    it "adds a new secondary content link if the base_path has a leading or trailing space" do
+      stub_user.permissions << "GDS Editor"
+
+      allow(Services.publishing_api).to receive(:lookup_content_id).and_return("a-content-id")
+      allow(Services.publishing_api).to receive(:get_content)
+        .with("a-content-id")
+        .and_return(content_item)
+      allow(Services.publishing_api).to receive(:put_content)
+      allow(Services.publishing_api).to receive(:lookup_content_ids).and_return([])
```

Comment:
Could this block be moved up to a context that is shared by the `it "adds a new secondary content link" do` test?

```ruby
describe "#create" do
  context "The content item exists" do
    before do
      allow(Services.publishing_api).to receive(:lookup_content_id).and_return("a-content-id")
      allow(Services.publishing_api).to receive(:get_content)
        .with("a-content-id")
        .and_return(content_item)
      allow(Services.publishing_api).to receive(:put_content)
      allow(Services.publishing_api).to receive(:lookup_content_ids).and_return([])
    end

    it "adds a new secondary content link" do
      # etc
    end

    it "adds a new secondary content link if the base_path has a leading or trailing space" do
      # etc
    end
```

I think some of the tests below could benefit from this too.

Reply:
The `it "accepts a full url as the base_path"` can be in this context too 👍